lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -62,75 +66,179 @@ def __init__(self, config):
+            if input_params.get('genetic_code'):
+                genome_obj.code_table = input_params['genetic_code']
+            genome_objs.append(genome_obj)
+
+            # do the upload
+            files = self._find_input_files(input_directory)
+            consolidated_file = self._join_files_skip_empty_lines(files)
+            consolidated_files.append(consolidated_file)
+
+        # save files to shock mass
+        file_handles = self._save_files_to_blobstore(consolidated_files)
+
+        # write and save assembly file
+        assemblies_ref = self._save_assemblies(
+            workspace_id, consolidated_files, inputs, genome_objs
+        )
+
+        # get assembly data
+        assemblies_data = self._get_assemblies_data(assemblies_ref)
+
+        for idx, input_params in enumerate(inputs):
+
+            genome = self._parse_genbank(
+                consolidated_files[idx],
+                input_params,
+                genome_objs[idx],
+                assemblies_ref[idx],
+                assemblies_data[idx],
+                file_handles[idx],
+            )
+            if input_params.get('genetic_code'):
+                genome["genetic_code"] = input_params['genetic_code']
+            genome_data.append(genome)
 
-        result = self.gi.save_one_genome({
-            'workspace': params['workspace_name'],
-            'name': params['genome_name'],
-            'data': genome,
-            "meta": params['metadata'],
-        })
-        ref = f"{result['info'][6]}/{result['info'][0]}/{result['info'][4]}"
-        logging.info(f"Genome saved to {ref}")
+            # clear the temp directory
+            shutil.rmtree(input_directories[idx])
 
-        # 5) clear the temp directory
-        shutil.rmtree(input_directory)
+        # save genomes
+        results = self._save_genomes(
+            workspace_id, genome_names, genome_data, genome_meta
+        )
 
-        # 6) return the result
-        info = result['info']
-        details = {
-            'genome_ref': ref,
-            'genome_info': info
-        }
+        # return the result
+        details = [
+            {'genome_ref': _upa(result["info"]), 'genome_info': result["info"]}
+            for result in results
+        ]
 
         return details
 
-    @staticmethod
-    def validate_params(params):
-        if 'workspace_name' not in params:
-            raise ValueError('required "workspace_name" field was not defined')
+    def _save_genomes(
+        self,
+        workspace_id,
+        genome_names,
+        genome_data,
+        genome_meta,
+    ):
+        results = [
+            self.gi.save_one_genome(
+                {
+                    'workspace': workspace_id,
+                    'name': name,
+                    'data': data,
+                    "meta": meta,
+                }
+            ) for name, data, meta in zip(genome_names, genome_data, genome_meta)
```

Comment:
This should be batched as well

Reply:
Currently, there is no batch method for `save_one_genome`

Reply:
Does the genbank to genome batch method call this method?

Reply:
genbanks_to_genomes -> import_genbank_mass -> _import_genbank_mass -> _save_genomes

Reply:
then we need to change the code so we save the genome objects in bulk. Whether that involves save_one_genome or not is up to you

Reply:
so I was not wrong: https://kbase.slack.com/archives/C03FK9RKSBX/p1706222953786629?thread_ts=1706222453.702059&cid=C03FK9RKSBX

Reply:
That just links to the re_team channel for me

Reply:
Are you talking about this thread? https://app.slack.com/client/T026VDM4X/C03FK9RKSBX

If so I'm not quite sure what you mean. A save_genomes function in the API wouldn't be useful for us

Reply:
Maybe we should hop in zoom and talk through this

Reply:
This gets fixed in the next PR

Reply:
:+1: